File: solutions/cpp/1207-unique-number-of-occurrences.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        unordered_map<int, int> count;
        for (int num : arr) {
            count[num]++;
        }
        unordered_set<int> uniqueCount;
        for (auto& pair : count) {
            if (!uniqueCount.insert(pair.second).second) {
                return false;
            }
        }
        return true;
    }
};

File: solutions/java/1207-unique-number-of-occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        Map<Integer, Integer> count = new HashMap<>();
        for (int num : arr) {
            count.put(num, count.getOrDefault(num, 0) + 1);
        }
        Set<Integer> uniqueCount = new HashSet<>(count.values());
        return uniqueCount.size() == count.size();
    }
}
